src/Data/Path/Pathy.purs

```diff
@@ -478,6 +477,27 @@ module Data.Path.Pathy
   parseAbsDir :: String -> Maybe (AbsDir Unsandboxed)
   parseAbsDir = parsePath (const Nothing) (const Nothing) (const Nothing) Just
 
+  -- | Folds a `Path` into a value using a specified function and a provided
+  -- | default.
+  fold :: forall c a b s. (String -> c -> c) -> c -> Path a b s -> c
+  fold _ z (Current                 ) = z
+  fold _ z (Root                    ) = z
+  fold f z (ParentIn p              ) = fold f z p
+  fold f z (FileIn   p (FileName f')) = f f' (fold f z p)
+  fold f z (DirIn    p (DirName  f')) = f f' (fold f z p)
```

Comment:
I would have preferred `f` and `d` myself, but I stayed consistent with the rest of the module, where it's `f` everywhere.

If I rename, it'll have to be en masse.